modules/optimization/src/materials/MisfitReporterOffsetFunctionMaterial.C

```diff
@@ -0,0 +1,70 @@
+#include "MisfitReporterOffsetFunctionMaterial.h"
+#include "MooseError.h"
+#include "libmesh/int_range.h"
+#include "libmesh/libmesh_common.h"
+
+registerMooseObject("OptimizationApp", MisfitReporterOffsetFunctionMaterial);
+registerMooseObject("OptimizationApp", ADMisfitReporterOffsetFunctionMaterial);
+
+template <bool is_ad>
+InputParameters
+MisfitReporterOffsetFunctionMaterialTempl<is_ad>::validParams()
+{
+  InputParameters params = ReporterOffsetFunctionMaterial::validParams();
+  params.addClassDescription(
+      "Computes the misfit and misfit gradient materials for inverse optimizations problems.");
+
+  params.addRequiredCoupledVar("sim_variable",
+                               "Variable that is being used for the simulation variable.");
```

Comment:
What if there are more than one variable? Also is this always the forward one and not the adjoint ?